src/fillTank.test.js

```diff
@@ -1,11 +1,98 @@
 'use strict';
 
 describe('fillTank', () => {
-  // const { fillTank } = require('./fillTank');
+   const { fillTank } = require('./fillTank');
 
-  it('should ', () => {
+  it('should be declared ', () => {
+    expect(fillTank).toBeInstanceOf(Function);
+  });
+
+  it('should calculate full tank If the amount is not given', () => {
+    const customer = {
+      money: 3000, // customer account balance
+      vehicle: {
+        maxTankCapacity: 40, // fuel tank volume
+        fuelRemains: 0, // Remaining fuel in the tank
+      },
+    };
+
+    fillTank(customer, 30);
+
+    expect(customer.money).toBe(1800);
+    expect(customer.vehicle.fuelRemains).toBe(40);
+  });
+
+  it('should calculate full tank If the amount is greater than the tank can accommodate', () => {
+    const customer = {
+      money: 3000, // customer account balance
+      vehicle: {
+        maxTankCapacity: 20, // fuel tank volume
+        fuelRemains: 5, // Remaining fuel in the tank
+      },
+    };
+
+    fillTank(customer, 50, 20);
+
+    expect(customer.money).toBe(2250);
+    expect(customer.vehicle.fuelRemains).toBe(20);
+  });
+
+  it('should fill the fuel if the client can pay for it', () => {
```

Comment:
The test description mentions filling the fuel if the client can pay for it, but the expected fuel remains is set to 20, which is less than the max capacity. Ensure that the test correctly reflects the scenario where the client can pay for the requested amount.